planner/core/find_best_task.go

```diff
@@ -196,6 +197,136 @@ func (ds *DataSource) tryToGetDualTask() (task, error) {
 	return nil, nil
 }
 
+// candidatePath is used to maintain required info for skyline pruning.
+type candidatePath struct {
+	path       *accessPath
+	columnSet  *intsets.Sparse // columnSet is the set of columns that occurred in the access conditions.
+	singleScan bool
+	matchProp  bool
+}
+
+// compareColumnSet will compares the two set. The last return value is used to indicate
+// if they are comparable, it is false when both two sets have columns that do not occur in the other.
+func compareColumnSet(l, r *intsets.Sparse) (int, bool) {
+	lLen, rLen := l.Len(), r.Len()
+	if lLen <= rLen {
+		if isSubset := l.SubsetOf(r); !isSubset {
+			return 0, false
+		}
+		if lLen == rLen {
+			return 0, true
+		}
+		return -1, true
+	}
+	if isSubset := r.SubsetOf(l); !isSubset {
+		return 0, false
+	}
+	return 1, true
+}
+
+func compareBool(l, r bool) int {
+	if l == r {
+		return 0
+	}
+	if l == false {
+		return -1
+	}
+	return 1
+}
+
+// compareCandidates is the core of skyline pruning. It compares the two candidate paths on three dimensions:
+// the set of columns that occurred in the access condition, whether or not it matches the physical property
```

Comment:
Make the format like this will be clearer.
1. the set of columns that occurred in the access condition
2. whether or not it matches the physical property
3. does it require a double scan